docker/build.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# This script creates builds according to our custom build policy, which is:
+# - "master" will always point to the most recent build on master
+# - "latest" will refer to the latest tagged release
+# - Each release "foo" will have a tag "foo"
+#
+# This script is a noop if HEAD is not origin/master OR tagged.
+
+set -exo pipefail
+
+function cleanup() {
+  docker system prune -f
+  find /tmp -name '*m3-docker' -print0 | xargs -0 rm -fv
+}
+
+trap cleanup EXIT
+
+function push() {
```

Comment:
nit: mind calling this pushimage or something longer than just push. The usage of this in the loop made me think it was related to `pushd` until I remembered this decl